development/compiling/compiling_for_x11.rst

```diff
@@ -169,3 +174,21 @@ here:
 You don't even need to copy them, you can just reference the resulting
 files in the ``bin/`` directory of your Godot source folder, so the next
 time you build, you automatically have the custom templates referenced.
+
+Using Clang and LLD for faster development
+------------------------------------------
+
+You can also use Clang and LLD to build Godot. This has two upsides compared to
+the default GCC + GNU ld setup:
+
+- LLD links Godot significantly faster compared to GNU ld or gold. This leads to
+  faster iteration times.
+- Clang tends to give more useful error messages compared to GCC.
```

Comment:
That's quite political :)

GCC warnings have improved a ton since GCC 6 and in my experience both GCC and Clang provide great warnings, sometimes on different codes. I wouldn't say that any is clearly better than the other in this regard.

Reply:
True, I based myself on how the situation was in 2016 or so. It's probably better now and we can remove that line.